src/overlays/actors/ovl_Bg_Jya_Lift/z_bg_jya_lift.c

```diff
@@ -26,24 +34,129 @@ const ActorInit Bg_Jya_Lift_InitVars = {
+    ICHAIN_F32(unk_FC, 2500, ICHAIN_STOP),
+};
+
+extern s32 D_0600D7E8;
+extern Gfx D_0600CCE0[];
+
+void BgJyaLift_InitDynapoly(BgJyaLift* this, GlobalContext* globalCtx, u32 arg2, DynaPolyMoveFlag moveFlag) {
+    s32 pad1;
+    u32 localConst = 0;
+
+    DynaPolyInfo_SetActorMove(&this->dyna, moveFlag);
+    DynaPolyInfo_Alloc(arg2, (void*)&localConst);
+    this->dyna.dynaPolyId = DynaPolyInfo_RegisterActor(globalCtx, &globalCtx->colCtx.dyna, &this->dyna, localConst);
+}
+
+void BgJyaLift_Init(Actor* thisx, GlobalContext* globalCtx) {
+    BgJyaLift* this = THIS;
+    this->unk_16A = 0;
+
+    if ((D_8089A020)) {
+        Actor_Kill(thisx);
+        return;
+    }
+
+    // Goddess lift CT
+    osSyncPrintf("女神リフト CT\n");
+    BgJyaLift_InitDynapoly(this, globalCtx, &D_0600D7E8, 0);
+    Actor_ProcessInitChain(thisx, sInitChain);
+    if (Flags_GetSwitch(globalCtx, (thisx->params & 0x3F))) {
+        BgJyaLift_SetFinalPosY(thisx);
+    } else {
+        BgJyaLift_SetInitPosY(thisx);
+    }
+    thisx->room = -1;
+    D_8089A020 = 1;
+    this->unk_16A = 1;
+}
+
+void BgJyaLift_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    BgJyaLift* this = THIS;
+
+    if (this->unk_16A != 0) {
+
+        // Goddess Lift DT
+        osSyncPrintf("女神リフト DT\n");
+        D_8089A020 = 0;
+        DynaPolyInfo_Free(globalCtx, &globalCtx->colCtx.dyna, this->dyna.dynaPolyId);
+    }
+}
+
+void BgJyaLift_SetInitPosY(BgJyaLift* this) {
+    this->actionFunc = func_80899D38;
+    this->dyna.actor.posRot.pos.y = 1613.0f;
+    this->unk_168 = 0;
+}
+
+void func_80899D38(BgJyaLift* this, GlobalContext* globalCtx) {
+    if ((Flags_GetSwitch(globalCtx, (this->dyna.actor.params & 0x3F)) != 0) || (this->unk_168 > 0)) {
+        this->unk_168++;
+        if (this->unk_168 >= 0x14) {
+            func_800800F8(globalCtx, 0xD66, -0x63, &this->dyna.actor, 0);
+            BgJyaLift_SetupMove(this);
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Jya_Lift/BgJyaLift_Destroy.s")
+void BgJyaLift_SetupMove(BgJyaLift* this) {
+    this->actionFunc = BgJyaLift_Move;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Jya_Lift/func_80899D18.s")
+void BgJyaLift_Move(BgJyaLift* this, GlobalContext* globalCtx) {
+    f32 distFromBottom;
+    f32 tempVelocity;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Jya_Lift/func_80899D38.s")
+    Math_SmoothScaleMaxMinF(&this->dyna.actor.velocity.y, 4.0f, 0.1f, 1.0f, 0.0f);
+    tempVelocity = (this->dyna.actor.velocity.y < 0.2f) ? 0.2f : this->dyna.actor.velocity.y;
+    distFromBottom = Math_SmoothScaleMaxMinF(&this->dyna.actor.posRot.pos.y, 973.0f, 0.1f, tempVelocity, 0.2f);
+    if ((this->dyna.actor.posRot.pos.y < 1440.0f) && (1440.0f <= this->dyna.actor.pos4.y)) {
+        func_8005B1A4(ACTIVE_CAM);
+    }
+    if (fabsf(distFromBottom) < 0.001f) {
+        BgJyaLift_SetFinalPosY(this);
+        Audio_PlayActorSound2(&this->dyna.actor, 0x287A);
```

Comment:
last thing forreal this time (sorry i missed this before lmao)
this sound effect is NA_SE_EV_ELEVATOR_STOP. you can find all of our sound effects defined in sfx.h
not every sound is going to have a define, so only use them when one exists